### PR TITLE
Refactor: Removing DatabaseBranches.

### PR DESCRIPTION
**Pull request by pimotte**
DatabaseBranches was simply a wrapper around a map providing some functionality that has been added to maps in Java 8.

### CODE REVIEW

**Comment by gtrefs** [comment]
Though, I am not a regular committer or contributer, I would like to give some comments here. If you don't think they are valid, just ignore them.

* You lose domain information
  * DatabaseBranches is a domain term and has some specific meaning. I.e. it is a collection of database branches. It hides specific technical details. By removing you lose this information.
* You don't own TreeMap and Map
  * You are not able to add more methods to TreeMap. You lose this ability by removing DatabaseBranches.
  * In the future, you might want to change from TreeMap to HashMap or another implemenation. With DatabaseBranches this is just an internal detail. Without, you need to change an awful lot of places. 

My conclusion and recommendation: Don't remove DatabaseBranches.

File: syncany-lib/src/it/java/org/syncany/tests/integration/database/DatabaseReconciliatorTest.java [hunk 1]
```diff
@@ -704,10 +705,14 @@ private void testFromMachinePerspective(String localMachineName, DatabaseVersion
 		TestResult actualTestResult = new TestResult();
 
 		// Get 'local' branch
-		DatabaseBranch localBranch = allBranches.getBranch(localMachineName);
+		DatabaseBranch localBranch = allBranches.get(localMachineName);
 
 		// Get all the other ones (clone 'all', and remove local)
-		DatabaseBranches unstitchedRemoteBranches = allBranches.clone();
+		Map<String, DatabaseBranch> unstitchedRemoteBranches = new TreeMap<String, DatabaseBranch>(){{
```
Comment (gtrefs) [inline syncany-lib/src/it/java/org/syncany/tests/integration/database/DatabaseReconciliatorTest.java:711]: Double curly braces are considered an anti pattern: https://blog.jooq.org/2014/12/08/dont-be-clever-the-double-curly-braces-anti-pattern/

See my comment on the conversation page. You alreday lost some information. While `allBranches.clone()` is quite readable and understandable, you needed to take a detour here by using the curly braces pattern.

File: syncany-lib/src/it/java/org/syncany/tests/integration/database/DatabaseReconciliatorTest.java [hunk 2]
```diff
@@ -736,10 +741,10 @@ private void testFromMachinePerspective(String localMachineName, DatabaseVersion
 				actualTestResult.winnersLastDatabaseVersionHeader);
 	}
 
-	private void printBranches(DatabaseBranches branches) {
-		for (String machineName : branches.getClients()) {
```
Comment (gtrefs) [inline syncany-lib/src/it/java/org/syncany/tests/integration/database/DatabaseReconciliatorTest.java:740]: Again `getClients` conveys much more information compared to `keySet`

**Comment by pimotte** [comment]
Thank you very much for your review!

I'd already started to get doubts when I was looking at what to tackle next, and I realized that "DatabaseBranch" is "just" a wrapper around an ArrayList, but refactoring it would make the code completely unreadable. 

It might be cleaner to refactor this as "extends Treemap<String, DatabaseVersionHeader>", but since that exposes a ton of methods that we don't necessarily need and might invalidate our assumptions on the data structures, so I don't think this is the way to go either. 

You've also convinced me this is certainly not the way to go either, so I'm closing this PR. Thank you! :)

**Pull request closed without merging**